core/commands/resolve.go

```diff
@@ -135,9 +135,11 @@ Resolve the value of an IPFS DAG path:
 			return err
 		}
 
-		c := rp.Cid()
+		if rp.Remainder() != "" {
+			return fmt.Errorf("path does not end on a dag-node boundary")
```

Comment:
We may want to keep the original error too because https://github.com/ipfs/interface-ipfs-core/pull/385/files#diff-8ce9a93c2e33e2c37099701b755064a3L85

Reply:
i.e. `found non-link at given path`

Reply:
That error message is just a side-effect of DagCBOR. But yeah, we can keep it for now.